src/resolvers/default.rs

```diff
@@ -170,15 +176,17 @@ impl Cipher for CipherChaChaPoly {
         copy_slices!(&nonce.to_le_bytes(), &mut nonce_bytes[4..]);
 
         let mut buf = Cursor::new(out);
-        let result = chacha20_poly1305_aead::decrypt(
-            &self.key,
-            &nonce_bytes,
+        let mut buffer = ciphertext[..ciphertext.len()-TAGLEN].to_vec();
+        let result = ChaCha20Poly1305::new(self.key.into()).decrypt_in_place_detached(
+            &nonce_bytes.into(),
             authtext,
-            &ciphertext[..ciphertext.len()-TAGLEN],
-            &ciphertext[ciphertext.len()-TAGLEN..],
-            &mut buf);
+            &mut buffer,
+            ciphertext[ciphertext.len()-TAGLEN..].into(),
+        );
         match result {
             Ok(_) => {
+                buf.write(&buffer).unwrap();
```

Comment:
Can we just do without this `Cursor` now that the new API doesn't demand it? Ideally we can avoid any heap allocations in this method.

Reply:
We need to pass in a type that implements [`aeda::Buffer`[(https://docs.rs/aead/0.2.0/aead/trait.Buffer.html) and I'm not sure we can without a `Vec`.

Reply:
No, I've got it actually